Security/Firewall/OAuthListener.php

```diff
@@ -29,34 +30,47 @@
 class OAuthListener implements ListenerInterface
 {
     /**
-     * @var \Symfony\Component\Security\Core\SecurityContextInterface
+     * @var SecurityContextInterface
+     *
+     * @deprecated since 1.4.3, to be removed in 2.0
      */
     protected $securityContext;
 
     /**
-     * @var \Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface
+     * @var TokenStorageInterface|SecurityContextInterface
+     */
+    protected $tokenStorage;
```

Comment:
Good idea, will work on it.

BTW, if we don't want override, class should be final later. ;-)